### PR TITLE
[17.12] bump containerd 1.0.2

### PR DESCRIPTION
**Pull request by cpuguy83**
Backport containerd 1.0.2 bump - https://github.com/moby/moby/pull/36308
```
git cherry-pick -s -x -Xsubtree=components/engine c2fb6db55be08da95b15bee730a191094f846577
```

Also bumps vendors to match contained 1.0.2 vendors... these don't match moby/moby since moby/moby is not tracking 1.0.2 for vendored deps.

### CODE REVIEW

**Comment by thaJeztah** [comment]
```
16:08:46 # github.com/docker/docker/daemon/graphdriver/graphtest
16:08:46 daemon/graphdriver/graphtest/graphbench_unix.go:248:13: undefined: driver.ReadFile
16:08:46 daemon/graphdriver/graphtest/testutil.go:38:12: undefined: driver.WriteFile
16:08:46 daemon/graphdriver/graphtest/testutil.go:44:12: undefined: driver.WriteFile
16:08:46 daemon/graphdriver/graphtest/testutil.go:48:9: undefined: driver.WriteFile
16:08:46 daemon/graphdriver/graphtest/testutil.go:58:22: undefined: driver.ReadFile
16:08:46 daemon/graphdriver/graphtest/testutil.go:77:9: undefined: driver.WriteFile
16:08:46 daemon/graphdriver/graphtest/testutil.go:135:14: undefined: driver.WriteFile
16:08:46 daemon/graphdriver/graphtest/testutil.go:170:15: undefined: driver.WriteFile
16:08:46 daemon/graphdriver/graphtest/testutil.go:177:15: undefined: driver.WriteFile
16:08:46 daemon/graphdriver/graphtest/testutil.go:206:24: undefined: driver.ReadFile
16:08:46 daemon/graphdriver/graphtest/testutil.go:206:24: too many errors
```

File: components/engine/vendor.conf [hunk 1]
```diff
@@ -104,15 +104,15 @@ github.com/googleapis/gax-go da06d194a00e19ce00d9011a13931c3f6f6887c7
 google.golang.org/genproto d80a6e20e776b0b17a324d0ba1ab50a39c8e8944
 
 # containerd
-github.com/containerd/containerd 89623f28b87a6004d4b785663257362d1658a729 # v1.0.0
+github.com/containerd/containerd cfd04396dc68220d1cecbe686a6cc3aa5ce3667c # v1.0.2
 github.com/containerd/fifo fbfb6a11ec671efbe94ad1c12c2e98773f19e1e6
-github.com/containerd/continuity 35d55c5e8dd23b32037d56cf97174aff3efdfa83
-github.com/containerd/cgroups 29da22c6171a4316169f9205ab6c49f59b5b852f
+github.com/containerd/continuity cf279e6ac893682272b4479d4c67fd3abf878b4e
```
Comment (thaJeztah) [inline components/engine/vendor.conf]: this one was already 4 commits ahead; https://github.com/containerd/continuity/compare/cf279e6ac893682272b4479d4c67fd3abf878b4e...35d55c5e8dd23b32037d56cf97174aff3efdfa83

File: components/engine/vendor.conf [hunk 2]
```diff
@@ -104,15 +104,15 @@ github.com/googleapis/gax-go da06d194a00e19ce00d9011a13931c3f6f6887c7
...
+github.com/containerd/cgroups c0710c92e8b3a44681d1321dcfd1360fc5c6c089
 github.com/containerd/console 84eeaae905fa414d03e07bcd6c8d3f19e7cf180e
-github.com/containerd/go-runc ed1cbe1fc31f5fb2359d3a54b6330d1a097858b7
+github.com/containerd/go-runc 4f6e87ae043f859a38255247b49c9abc262d002f
 github.com/containerd/typeurl f6943554a7e7e88b3c14aad190bf05932da84788
 github.com/dmcgowan/go-tar go1.10
-github.com/stevvooe/ttrpc 76e68349ad9ab4d03d764c713826d31216715e4f
+github.com/stevvooe/ttrpc d4528379866b0ce7e9d71f3eb96f0582fc374577
```
Comment (thaJeztah) [inline components/engine/vendor.conf]: full diff: https://github.com/stevvooe/ttrpc/compare/76e68349ad9ab4d03d764c713826d31216715e4f...d4528379866b0ce7e9d71f3eb96f0582fc374577

File: components/engine/vendor.conf [hunk 3]
```diff
@@ -104,15 +104,15 @@ github.com/googleapis/gax-go da06d194a00e19ce00d9011a13931c3f6f6887c7
...
-github.com/containerd/containerd 89623f28b87a6004d4b785663257362d1658a729 # v1.0.0
+github.com/containerd/containerd cfd04396dc68220d1cecbe686a6cc3aa5ce3667c # v1.0.2
 github.com/containerd/fifo fbfb6a11ec671efbe94ad1c12c2e98773f19e1e6
-github.com/containerd/continuity 35d55c5e8dd23b32037d56cf97174aff3efdfa83
-github.com/containerd/cgroups 29da22c6171a4316169f9205ab6c49f59b5b852f
+github.com/containerd/continuity cf279e6ac893682272b4479d4c67fd3abf878b4e
+github.com/containerd/cgroups c0710c92e8b3a44681d1321dcfd1360fc5c6c089
 github.com/containerd/console 84eeaae905fa414d03e07bcd6c8d3f19e7cf180e
-github.com/containerd/go-runc ed1cbe1fc31f5fb2359d3a54b6330d1a097858b7
+github.com/containerd/go-runc 4f6e87ae043f859a38255247b49c9abc262d002f
```
Comment (thaJeztah) [inline components/engine/vendor.conf]: full diff: https://github.com/containerd/go-runc/compare/ed1cbe1fc31f5fb2359d3a54b6330d1a097858b7...4f6e87ae043f859a38255247b49c9abc262d002f

File: components/engine/vendor.conf [hunk 4]
```diff
@@ -104,15 +104,15 @@ github.com/googleapis/gax-go da06d194a00e19ce00d9011a13931c3f6f6887c7
 google.golang.org/genproto d80a6e20e776b0b17a324d0ba1ab50a39c8e8944
 
 # containerd
-github.com/containerd/containerd 89623f28b87a6004d4b785663257362d1658a729 # v1.0.0
+github.com/containerd/containerd cfd04396dc68220d1cecbe686a6cc3aa5ce3667c # v1.0.2
 github.com/containerd/fifo fbfb6a11ec671efbe94ad1c12c2e98773f19e1e6
-github.com/containerd/continuity 35d55c5e8dd23b32037d56cf97174aff3efdfa83
-github.com/containerd/cgroups 29da22c6171a4316169f9205ab6c49f59b5b852f
+github.com/containerd/continuity cf279e6ac893682272b4479d4c67fd3abf878b4e
+github.com/containerd/cgroups c0710c92e8b3a44681d1321dcfd1360fc5c6c089
```
Comment (thaJeztah) [inline components/engine/vendor.conf]: full diff: https://github.com/containerd/cgroups/compare/29da22c6171a4316169f9205ab6c49f59b5b852f...c0710c92e8b3a44681d1321dcfd1360fc5c6c089

File: components/engine/vendor.conf [hunk 5]
```diff
@@ -104,15 +104,15 @@ github.com/googleapis/gax-go da06d194a00e19ce00d9011a13931c3f6f6887c7
 google.golang.org/genproto d80a6e20e776b0b17a324d0ba1ab50a39c8e8944
 
 # containerd
-github.com/containerd/containerd 89623f28b87a6004d4b785663257362d1658a729 # v1.0.0
+github.com/containerd/containerd cfd04396dc68220d1cecbe686a6cc3aa5ce3667c # v1.0.2
```
Comment (thaJeztah) [inline components/engine/vendor.conf]: full diff: https://github.com/containerd/containerd/compare/89623f28b87a6004d4b785663257362d1658a729...cfd04396dc68220d1cecbe686a6cc3aa5ce3667c

File: components/engine/vendor.conf (hunk 1 above)
Comment (cpuguy83) [reply components/engine/vendor.conf]: I removed this change.

**Comment by thaJeztah** [comment]
`DockerSwarmSuite.TestAPISwarmServicesUpdateStartFirst` failing in CE and s390x; Is that a flaky test?

https://jenkins.dockerproject.org/job/docker-ce-pr/611/execution/node/748/log/

```
21:20:52 ----------------------------------------------------------------------
21:20:52 FAIL: check_test.go:366: DockerSwarmSuite.TearDownTest
21:20:52 
21:20:52 check_test.go:371:
21:20:52     d.Stop(c)
21:20:52 daemon/daemon.go:395:
21:20:52     t.Fatalf("Error while stopping the daemon %s : %v", d.id, err)
21:20:52 ... Error: Error while stopping the daemon de1035ae49ef1 : exit status 130
21:20:52 
21:20:52 
21:20:52 ----------------------------------------------------------------------
21:20:52 PANIC: docker_api_swarm_service_test.go:201: DockerSwarmSuite.TestAPISwarmServicesUpdateStartFirst
21:20:52 
21:20:52 [de1035ae49ef1] waiting for daemon to start
21:20:52 [de1035ae49ef1] daemon started
21:20:52 
21:20:52 [de1035ae49ef1] daemon started
21:20:52 Attempt #2: daemon is still running with pid 32117
21:20:52 Attempt #3: daemon is still running with pid 32117
21:20:52 Attempt #4: daemon is still running with pid 32117
21:20:52 [de1035ae49ef1] exiting daemon
21:20:52 ... Panic: Fixture has panicked (see related PANIC)
21:20:52 
```

https://jenkins.dockerproject.org/job/docker-ce-pr-s390x/406/execution/node/458/log/

```
21:14:54 ----------------------------------------------------------------------
21:14:54 FAIL: check_test.go:366: DockerSwarmSuite.TearDownTest
21:14:54 
21:14:54 check_test.go:371:
21:14:54     d.Stop(c)
21:14:54 daemon/daemon.go:395:
21:14:54     t.Fatalf("Error while stopping the daemon %s : %v", d.id, err)
21:14:54 ... Error: Error while stopping the daemon d94caeef2a019 : exit status 130
21:14:54 
21:14:54 
21:14:54 ----------------------------------------------------------------------
21:14:54 PANIC: docker_api_swarm_service_test.go:201: DockerSwarmSuite.TestAPISwarmServicesUpdateStartFirst
21:14:54 
21:14:54 [d94caeef2a019] waiting for daemon to start
21:14:54 [d94caeef2a019] daemon started
21:14:54 
21:14:54 [d94caeef2a019] daemon started
21:14:54 Attempt #2: daemon is still running with pid 300
21:14:54 Attempt #3: daemon is still running with pid 300
21:14:54 Attempt #4: daemon is still running with pid 300
21:14:54 [d94caeef2a019] exiting daemon
21:14:54 ... Panic: Fixture has panicked (see related PANIC)
21:14:54 
```

**Comment by cberner** [comment]
@cpuguy83 , @thaJeztah just checking in. Will this be in 17.12.1?

**Comment by cpuguy83** [comment]
@thaJeztah I don't know, how do we rebuild?

@cberner If we can resolve issues in time.

**Comment by cpuguy83** [comment]
I have requested that this be considered a p0 for 1.12.1.

**Comment by thaJeztah** [comment]
@seemethere can you restart CI?

**Comment by seemethere** [comment]
@thaJeztah done

**Comment by sebbrandt87** [comment]
What about the failing ce-tests-s390x?

**Comment by cpuguy83** [comment]
@seemethere

**Comment by sebbrandt87** [comment]
```
Cannot contact s390x-ubuntu-22: java.io.IOException: Remote call on JNLP4-connect connection from ip-172-31-0-7.us-west-2.compute.internal/172.31.0.7:18173 failed
```

**Comment by thaJeztah** [comment]
Looks like this test is flaky `DockerSwarmSuite.TestAPISwarmServicesUpdateStartFirst`

```
21:52:17 PASS: docker_api_swarm_service_test.go:141: DockerSwarmSuite.TestAPISwarmServicesUpdate	31.527s
21:52:58 
21:52:58 ----------------------------------------------------------------------
21:52:58 FAIL: check_test.go:366: DockerSwarmSuite.TearDownTest
21:52:58 
21:52:58 check_test.go:371:
21:52:58     d.Stop(c)
21:52:58 daemon/daemon.go:395:
21:52:58     t.Fatalf("Error while stopping the daemon %s : %v", d.id, err)
21:52:58 ... Error: Error while stopping the daemon d59e6f1144b65 : exit status 130
21:52:58 
21:52:58 
21:52:58 ----------------------------------------------------------------------
21:52:58 PANIC: docker_api_swarm_service_test.go:201: DockerSwarmSuite.TestAPISwarmServicesUpdateStartFirst
21:52:58 
21:52:58 [d59e6f1144b65] waiting for daemon to start
21:52:58 [d59e6f1144b65] daemon started
21:52:58 
21:52:58 [d59e6f1144b65] daemon started
21:52:58 Attempt #2: daemon is still running with pid 839
21:52:58 Attempt #3: daemon is still running with pid 839
21:52:58 Attempt #4: daemon is still running with pid 839
21:52:58 [d59e6f1144b65] exiting daemon
21:52:58 ... Panic: Fixture has panicked (see related PANIC)
21:52:58 
```

**Comment by sebbrandt87** [comment]
I guess the plan is now to fix the flaky test?

**Comment by cpuguy83** [comment]
I think that is a legit swarmkit problem, which seems to be easily on s390x. Hitting it on moby/moby as well.

**Comment by cpuguy83** [comment]
Well, 17.12.1 is already cut.

**Comment by cpuguy83** [comment]
Closing since this is passed the 17.12 support window.

**Pull request closed without merging**